dip-0031.md

```diff
@@ -0,0 +1,120 @@
+masternode, will result in a masternode voting against the target in all future DKG sessions until
+that targeted masternode has been successfully banned.
+
+## Motivation
+
+To ensure that evonodes provide service on Dash Platform and that non-functioning evonodes do not
+participate in the Platform validator set, Platform needs a way to exclude faulty evonodes from
+future Platform validator sets / quorums. Since these nodes are not properly providing service, they
+should be proof-of-service (PoSe) banned. This DIP outlines a method for Platform to securely notify
+Core that a node should be banned.
+
+## Prior Work
+
+* [DIP-0006: Long-Living Masternode
+  Quorums](https://github.com/dashpay/dips/blob/master/dip-0006.md)  
+* [DIP-0007: LLMQ Signing Requests /
+  Sessions](https://github.com/dashpay/dips/blob/master/dip-0007.md)
+
+## PlatformBan P2P Message
+
+Core will introduce a new P2P message, which will facilitate the PoSe banning of nodes identified as
+not providing proper service on Platform. This P2P message will be broadcast via the inventory
+system and introduced in Protocol Version `70235`.
+
+| Field | Type | Size(bytes) | Description |
+| ----- | ----- | ----- | ----- |
+| protxHash | uint256 | 32 | The protxHash of the masternode being banned |
+| signedHeight | int32_t | 4 | The height at which this message was signed by the platform quorum |
+| quorumHash | uint256 | 32 | The quorumHash of the quorum which signed the message |
+| blsSignature | byte[] | 96 | The BLS signature over the message, by the quorum identified by quorumHash |
+
+Note: this P2P message intentionally excludes an internal version; instead, it is versioned via the
+Protocol Version.
+
+## Validation
+
+When received, a node must validate the message. It should do this by:
+
+1. Check if the node is synced  
+   * Penalty: skip validation of message, do not forward, do not mark as received in inventory  
+2. Validate the protxHash is in the masternodeList  
+   * Penalty: small P2P penalty (1), as the masternode may have very recently been removed  
+3. Validate the masternode associated with the protxHash is an evonode and not a regular masternode  
+   * Penalty: Ban node, P2P penalty (100)  
+4. Validate the signedHeight is valid and not too far in the past or future  
+   * Penalty: medium P2P penalty (10), it is possible we think we are synced, when we are not  
+5. Validate the quorumHash, with quorum type equal to the platform quorum, exists at the
+   signedHeight  
+   * Penalty: Ban node, P2P penalty (100)  
+6. Validate the blsSignature is valid for the quorumHash over the protxHash and signedHeight  
+   * Penalty: Ban node, P2P penalty (100)
+
+The inventory system should mark the message as received unless validation case 1 fails.
+
+## Processing
+
+Once validation succeeds, Core shall adjust an internal `PlatformPoSeBan` flag from `false` to
+`true`. This flag should be stored on disk and persisted across restarts.
```

Comment:
We need to be more precise on where the internal `PlatformPoSeBan` flag should be.
I guess the most appropriate would be `CDeterministicMNState`.

Reply:
This is an implementation detail; not strictly relevant for DIP imo

Reply:
Agree with @PastaPastaPasta on this one